src/ray/protobuf/gcs_service.proto

```diff
@@ -437,6 +437,11 @@ enum GcsServiceFailureType {
   GCS_SERVER_RESTART = 1;
 }
 
+enum FilterPredicate {
+  EQUAL = 0;
```

Comment:
nit: protobuf encourages to declare unspecified default value: https://protobuf.dev/programming-guides/dos-donts/#unspecified-enum, though for us it's less of a problem since we don't have different releases between client and server

Reply:
Good point! Will add a default value. 

At the same time, I would think the backward compatibility is already broken when the format of the request is updated. And I'm hoping the format change in this PR can be extensible enough so that all the future updates can backward compatible changes.

Reply:
we don't need backward compatibility in this. We do need to know if a filter is "not set" or "EQUAL" but we already can test that. So I guess current enum is OK

Reply:
That's right. I think backward compatibility is not needed for now. 

At the same time, if ever we want to separate different components (API server and GCS in this case) into different products, compatibility will be a concern at that time.